src/ComputeSharp.D2D1/Shaders/Interop/Effects/PixelShaderEffect.ID2D1DrawTransformMethods.cs

```diff
@@ -211,16 +225,41 @@ public static int MapInputRectsToOutputRect(PixelShaderEffect* @this, RECT* inpu
                 // transform. In this case, the target area will be the output node anyway.
                 outputRect->MakeD2D1Infinite();
 
-                @this->inputRect = default;
-
                 *outputOpaqueSubRect = default;
             }
             else
             {
-                // Default mapping
-                *outputRect = inputRects[0];
+                // If at least one input is present and no custom mapper is available, apply the default
+                // mapping. In this case, the output rect should be the union of the input rects for all
+                // the simple shader inputs, or infinite if there are no simple inputs. The input rects
+                // for complex inputs, if present, are not needed in this scenario, so they're ignored.
+                RECT unionOfSimpleRects = default;
+                bool isUnionOfSimpleRectsEmpty = true;
```

Comment:
or, you know, you could use `RECT?` :) same outcome, of course

Reply:
It was really annoying me that `T?.Value` is a property, so I'd end up with more extra copies 😆